test/cap-automator/index.test.ts

```diff
@@ -0,0 +1,552 @@
+    const wstethWhale = '0x5fEC2f34D80ED82370F733043B6A536d7e9D7f8d' as const
+
+    const deposit = async (signerAddress: string, tokenAddress: string, amount: BigInt) => {
+        await impersonateAccount(signerAddress)
+        const signer = await hre.ethers.getSigner(signerAddress)
+
+        const token = new Contract(tokenAddress, erc20Abi, signer)
+
+        await token.approve(addresses.mainnet.pool, amount)
+
+        const pool = new Contract(addresses.mainnet.pool, poolAbi, signer)
+        await pool.supply(tokenAddress, amount, signerAddress, 0)
+    }
+
+    const borrow = async (signerAddress: string, tokenAddress: string, amount: BigInt) => {
+        await impersonateAccount(signerAddress)
+        const signer = await hre.ethers.getSigner(signerAddress)
+
+        const pool = new Contract(addresses.mainnet.pool, poolAbi, signer)
+        await pool.borrow(tokenAddress, amount, 2, 0, signerAddress)
+    }
+
+    const formatExecSupplyCallData = (assetAddress: string) =>
+        `0xb00d4b1c000000000000000000000000${assetAddress.slice(2).toLocaleLowerCase()}`
+
+    const formatExecBorrowCallData = (assetAddress: string) =>
+        `0xb1ae9f48000000000000000000000000${assetAddress.slice(2).toLocaleLowerCase()}`
+
+    const formatExecCallData = (assetAddress: string) =>
+        `0x6bb6126e000000000000000000000000${assetAddress.slice(2).toLocaleLowerCase()}`
+
+    before(async () => {
+        ;[reader, keeper] = await ethers.getSigners()
+
+        pool = new Contract(addresses.mainnet.pool, poolAbi, reader)
+        capAutomator = new Contract(addresses.mainnet.capAutomator, capAutomatorAbi, reader)
+        protocolDataProvider = new Contract(addresses.mainnet.protocolDataProvider, protocolDataProviderAbi, reader)
+
+        sparkAssets = await pool.getReservesList()
+
+        for (const assetAddress of sparkAssets) {
+            await capAutomator.exec(assetAddress)
+        }
+
+        await mine(2, { interval: 24 * 60 * 60 })
+
+        capAutomatorW3F = w3f.get('cap-automator')
+    })
+
+    beforeEach(async () => {
+        snapshotRestorer = await takeSnapshot()
+    })
+
+    afterEach(async () => {
+        await snapshotRestorer.restore()
+    })
+
+    it('no cap updates are required', async () => {
+        const { result } = await capAutomatorW3F.run('onRun')
+
+        expect(result.canExec).to.equal(false)
+        !result.canExec && expect(result.message).to.equal('No cap automator calls to be executed')
+    })
+
+    describe('execSupply', () => {
+        const testedThresholds = [3000, 5000, 7000, 9000]
+
+        testedThresholds.forEach((threshold) => {
+            describe(`${threshold / 100}% threshold`, () => {
+                it(`actual gap is smaller than optimal but the threshold is not met`, async () => {
+                    const { gap } = await capAutomator.supplyCapConfigs(wbtc)
+
+                    const percentageOfTheGapTooSmallToTrigger = (10000 - threshold) / 100 - 1
```

Comment:
This can be updated to be `const percentageOfTheGapTooSmallToTrigger = (10000 - threshold) / 100` for all `percentageOfTheGapTooSmallToTrigger` conditions